### PR TITLE
 Fix pre and post execution script parser 

### PR DESCRIPTION
**Pull request by yakutovicha**
Previously, lines containing '#' symbol where generating parsing errors
as well as the lines starting with '#='. Now it should be fixed: lines
starting with #= are completely ignored, while presence of '#' symbols
does not cause any errors.

Fixes the issue #1965

### CODE REVIEW

File: aiida/cmdline/utils/multi_line_input.py [hunk 1]
```diff
@@ -31,12 +32,38 @@ def edit_pre_post(pre=None, post=None, summary=None):
...
-        stripped_input = re.sub(r'(^#.*$\n)+', '#', mlinput, flags=re.M).strip('#')
-        pre, post = [text.strip() for text in stripped_input.split('#')]
+
+        # Splitting the text in pre- and post- halfs
+        try:
+            pre, post = mlinput.split(separator)
+        except ValueError as err:
+            if str(err) == "need more than 1 value to unpack":
+                raise InputValidationError("Looks like you modified the "
+                                           "separartor that should NOT be modified. Please be "
```
Comment (sphuber) [inline aiida/cmdline/utils/multi_line_input.py]: small typo `separartor`. We should also catch this `InputValidationError` in all the `verdi` commands that call `ensure_scripts` e.g. line 194 of `cmd_computer`. Wrap those in try/catch and raise `click.BadParameter` like so:
```
        try:
            pre, post = ensure_scripts(kwargs.pop('prepend_text', ''), kwargs.pop('append_text', ''), kwargs)
        except InputValidationError as exception:
            raise click.BadParameter('invalid prepend and or append text: {}'.format(exception))
```

File: aiida/cmdline/utils/multi_line_input.py [hunk 2]
```diff
@@ -31,12 +32,38 @@ def edit_pre_post(pre=None, post=None, summary=None):
     template = env.get_template('prepost.bash.tpl')
     summary = summary or {}
     summary = {k: v for k, v in summary.items() if v}
-    content = template.render(default_pre=pre or '', default_post=post or '', summary=summary)
+
+    # Define a separator that will be splitting pre- and post- execution
+    # parts of the submission script
+    separator = "#====================================================#\n" \
+                "#=               Post execution script              =#\n" \
+                "#=  I am acting a as separator, do not modify me!!! =#\n" \
```
Comment (sphuber) [inline aiida/cmdline/utils/multi_line_input.py]: you can add the article `a` but then it should come after `as`, i.e. it should be `I am acting as a separator`

Comment (yakutovicha) [reply aiida/cmdline/utils/multi_line_input.py]: of course, sorry :)

File: aiida/cmdline/utils/multi_line_input.py (hunk 1 above)
Comment (yakutovicha) [reply aiida/cmdline/utils/multi_line_input.py]: done.

**Comment by sphuber** [comment]
Thanks a lot @yakutovicha

**Pull request merged**